### PR TITLE
1531 inform the user about deprecation and upgrade options

### LINKED ISSUE #1531: Inform the user about deprecation and upgrade options
As a cobigen user, I want to get automatically informed about the conceptual changes of the configuration going from monolithic template configuration folder to a modular template-set configuration folder. I want to get offered information (#1516).

*Acceptance Criteria*
* [x] If cobigen reads my monolithic template configuration, which is deprecated, I want to get offered an upgrade option and get informed about the process and steps to be performed
* [x] To properly make my decision in the moment of time, I would like to get more information about why I should do the upgrade and what is the impact 
* [x] the user should be asked every time a monolithic configuration is read from a folder 
* [x] make sure, that maven execution is not blocked by this feature at all

### PR DESCRIPTION
**Pull request by mdukhan**
Adresses/Fixes #1531 

Implements

* Generation CLI DeprecatedMonolithicTemplatesException Handler 

@devonfw/cobigen

### CODE REVIEW

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 1]
```diff
@@ -121,12 +123,34 @@ public Integer doAction() throws Exception {
...
+   *
+   */
+  private void checkOldTemplatesException() {
+
+    try {
+      CobiGenUtils.initializeCobiGen(this.templatesProject);
+    } catch (DeprecatedMonolithicTemplatesException e) {
+      e.printStackTrace();
+      LOG.info("Would you like to upgrade your templates to the newest version? \n"
+          + "type yes/y to continue or no/n to cancel (or hit return for yes).", System.getProperty("user.dir"));
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:139]: We should remove the printStackTrace and add the exception to the info or maybe another warning message.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java [hunk 2]
```diff
@@ -121,12 +123,34 @@ public Integer doAction() throws Exception {
     return 0;
   }
 
+  /**
+   * check if Old templates exists to handle the exception and let the user decides, if the templates should be upgraded
+   *
+   *
+   *
+   */
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java:131]: Better description: Uses default initialization, checks if old templates exist, handles the exception and lets the user decide if the templates should be upgraded.
Extra returns can be removed too.

File: cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenUtils.java [hunk 3]
```diff
@@ -60,6 +60,22 @@ public static CobiGen initializeCobiGen(Path templatesProject) {
     }
   }
 
+  /**
+   * Registers CobiGen plug-ins and instantiates CobiGen, even if Old templates found
```
Comment (jan-vcapgemini) [inline cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenUtils.java:64]: Old can be lowercase.

**Review by jan-vcapgemini — CHANGES_REQUESTED** [review]

**Comment by mdukhan** [comment]
#

**Pull request closed without merging**